Source/WebCore/style/values/inline/StyleLineHeight.cpp

```diff
@@ -70,9 +68,19 @@ auto CSSValueConversion<LineHeight>::operator()(BuilderState& state, const CSSPr
 
     if (primitiveValue.isLength() || primitiveValue.isCalculatedPercentageWithLength()) {
         double fixedValue = 0;
-        if (primitiveValue.isLength())
+        if (primitiveValue.isLength()) {
+            auto lengthUnit = CSS::toLengthUnit(primitiveValue.primitiveType());
             fixedValue = primitiveValue.resolveAsLength(conversionData);
-        else
+
+            // Apply text zoom to font-relative units when evaluationTimeZoomEnabled.
+            // This matches the behavior of percentage-based line-height (see primitiveValue.isPercentage() case below).
+            // When evaluationTimeZoomEnabled is true, computedSizeForRangeZoomOption returns
+            // the unzoomed font size, so we need to multiply by text zoom for font-relative units.
+            if (lengthUnit && CSS::isFontOrRootFontRelativeLength(*lengthUnit)) {
+                auto textZoom = evaluationTimeZoomEnabled(state) ? conversionData.zoom() : 1.0f;
+                fixedValue *= textZoom;
```

Comment:
Maybe this entire block could be guarded by `evaluationTimeZoomEnabled(state)` and not need to multiply by 1.0 but also maybe this is more readable

Reply:
I've opted for this way for readability, to make sure the reader understands when text zoom is used. That said, I'm planning to refactor the whole thing soon to be resolved when resolving length instead of each individual property.